dvc/repo/get.py

```diff
@@ -49,16 +66,31 @@ def get(url, path, out=None, rev=None):
             # the same cache file might be used a few times in a directory.
             repo.cache.local.cache_types = ["reflink", "hardlink", "copy"]
 
-            o = repo.find_out_by_relpath(path)
+            output = None
+            output_error = None
+
+            try:
+                output = repo.find_out_by_relpath(path)
+            except OutputNotFoundError as ex:
+                output_error = ex
+
+            is_git_file = output_error and not os.path.isabs(path)
+            is_not_cached = output and not output.use_cache
```

Comment:
If it is not cached then this is also a git file, so this var names are confusing. Overall this logic is more complicated than necessary. It is simply either cached or a git managed file, so:

```python
try:
    if out and out.use_cache:
        # do the pull and checkout
        ...
    else:
        # Non cached out outside repo, can't be handled
        if os.path.abspath(src):
            raise PathOutsideRepoError(...)

        # it's git-handled and already checked out to tmp dir
        # we should just copy, not a git specific operation
        ...
        copy_dir_or_file(src_full, dst_full)  
except FileNotFoundError:
    raise FileMissingError(...)
```

And again forgot about generic exception, basically:
```console
$ dvc get http://some.repo non-existing-path
Can't find non-existing-path in some.repo neither as output 
nor as git handled file/dir
```

Message may be different, but the idea is that we don't know whether user tried to get an out or a git handled file.

Reply:
@Suor We've agreed to not use FileMissingError as its message is not applicable here. Hence `PathOutsideRepoError`, which is more suitable. The current logic corresponds to what we have in Repo.open.

Reply:
Agreed on the naming.

Reply:
And indeed output_error is not wrapped as it is in repo.open.

Reply:
Looking closely at Repo.open, it indeed has a more clear implementation than this, as it doesn't introduce is_git_file confusion.